File: week#4/task#2/Shapes.java
// main parent class
public class Shapes {
    // variables 
    static float PI = 3.14f;
    float radius;
    float height;
    float width;

    // main method
    public static void main(String[] args) {
    // creating an object of circle
        Circle shape1 = new Circle(10f);
    // calling shape1(circle) method
        System.out.println("Area of Circle is " + shape1.findArea());
    // creating an object of Rectangle
        Rectangle shape2 = new Rectangle(10f, 10f);
    // calling shape2(circle) method
        System.out.println("Area of Rectangle is " + shape2.findArea());
    }

}

// class method
class Circle extends Shapes{

    // constructor

    Circle(float radius) {
        this.radius = radius;
    }

    // method
    float findArea() {
        float area = (radius * radius) * PI;
        return area;
    }
    
}


class Rectangle extends Shapes{

    // constructor
    Rectangle(float height, float width) {
        this.height = height;
        this.width = width;
    }

    // method
    float findArea() {
        float area = height * width;
        return area;
    }
}

File: week#4/task#2/Shapes.cpp
// NOTE: OUTPUT IS ZERO: fix that

// header file
#include<iostream>
// using these for easy string & output
using std::string;
using std::cout;

// main/parent class
class Shapes {
// public modifier: everything is private by default in C++
public:
// members/attributes
    float radius;
// only declaring static as it cannot be initilized in class
    static float PI;
};
// initilized the static variables
float Shapes::PI = 3.14;

// child class
class Circle : public Shapes {
public:
// Circle constructor using attributes from parent
    Circle(float radius) {
    }
// findArea method
    void findArea() {
        float area = (radius * radius) * PI;
        cout<<"The area of Circle: "<<area;
    }
};

int main() {
    // creating the object
    Circle circle(10);
    // calling the method of the object
    circle.findArea();
}
